provider/kubernetes.go

```diff
@@ -170,10 +170,16 @@ func (provider *Kubernetes) loadIngresses(k8sClient k8s.Client) (*types.Configur
 					log.Errorf("Error retrieving services %s", pa.Backend.ServiceName)
 				}
 				for _, service := range services {
-					var protocol string
+					protocol := "http"
 					for _, port := range service.Spec.Ports {
 						if port.Port == pa.Backend.ServicePort.IntValue() {
-							protocol = port.Name
+							if len(port.Name) == 0 {
+								if port.Port == 443 {
+									protocol = "https"
+								}
+							} else {
+								protocol = port.Name
```

Comment:
Would it make sense to use the Ingress object instead of the Service to determine the protocol? ServicePorts can be named arbitrarily, while the IngressSpec states that SSL is only supported on port 443. All others are http.